proposals/web-api-add-edit-with-media.md

```diff
@@ -0,0 +1,618 @@
+
+**REQUEST**
+```http
+POST https://api.my-webapi.io/Media
+OData-Version: 4.01
+Content-Type: application/json
+Accept: application/json
+Prefer: return=representation
+```
+```json
+{
+ "Caption": "Ipsum Lorum",
+ "Order": 1,
+ "ImageType": "image/jpeg",
+ 
+}
+```
+
+**RESPONSE - External Media storage**
+```http
+HTTP/2 201 Created
+OData-Version: 4.01
+EntityId: "12345"
+Location: https://api.my-webapi.io/Media('12345')
+Content-Length: 200
+Content-Type: application/json
+Preference-Applied: return=representation
+```
+```json
+{
+  "@odata.context": "https://api.my-webapi.io/$metadata#Media/$entity",
+  "@odata.id": "Media('12345')",
+  "@odata.editLink": "https://api.my-webapi.io/Media('12345')",
+  "@odata.mediaReadLink": "https://api-my-webapi.io/image/not_available.jpg",
+  "@odata.mediaEditLink": "https://storage.my-webapi.io/media/12345.jpg?authentication_token=my-one-time-use-auth-token-12345-zyxwut",
+  "@odata.etag": "W/\"aBcDeFgHiJkLmNoPqRsTuVwXyz\"",
+  "MediaObjectID": "12345",
+  "Caption": "Ipsum Lorum",
+  "Order": 1,
+  "ImageType": "image/jpeg",
+  
+  "MediaStatus": "Incomplete",
+  "MediaStatusDescription": "Awaiting Byte Stream",
+}
+```
+
+**RESPONSE - Media through Web API**
+
+When the `MediaEditLink` annotation is not provided, there is an implicit URL for the editMediaLink `https://api.my-webapi.io/Media('12345')/$value` provided by the OData specification
+
+```http
+HTTP/2 201 Created
+OData-Version: 4.01
+EntityId: "12345"
+Location: https://api.my-webapi.io/Media('12345')
+Content-Length: 200
+Content-Type: application/json
+Preference-Applied: return=representation
+```
+```json
+{
+  "@odata.context":"https://api.my-webapi.io/$metadata#Media/$entity",
+  "@odata.id":"Media('12345')",
+  "@odata.editLink":"https://api.my-webapi.io/Media('12345')",
+  "@odata.mediaReadLink": "https://api-my-webapi.io/image/not_available.jpg",
+  "@odata.etag": "W/\"aBcDeFgHiJkLmNoPqRsTuVwXyz\"",
+  "MediaObjectID": "12345",
+  "Caption": "Ipsum Lorum",
+  "Order": 1,
+  "ImageType": "image/jpeg",
+  
+  "MediaStatus": "Incomplete",
+  "MediaStatusDescription": "Awaiting Byte Stream",
+}
+```
+
+The return from the POST will return the Media record with the object in an `Incomplete` status.  The next step in the process is to provide the byte stream for the media object to the server at the provided endpoint.  The client MUST use the `odata.mediaEditLink` if provided and use the implicit URL only if one is not provided as per the OData specification.
+
+### Upload the Media byte array
+
+The byte stream upload is a simple HTTP POST transaction to the provided to the endpoint.  If it is an request outside the WebApi domain, then all required data (authentication, etc.) MUST be provided in the `mediaEditLink` URL.  If `mediaEditLink` is not in WebApi domain, then the established authentication (cookies, authentication headers, etc.) MUST continue to be provided.
+
+**REQUEST - External Target**
```

Comment:
Instead of **REQUEST - External Target**

This should be put in its own section called: "Example: Media with External Target" with a one-sentence description of the use case and its own REQUEST and RESPONSE items, even if duplicative.